clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -1980,6 +1981,23 @@ static void handleWeakRefAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   D->addAttr(::new (S.Context) WeakRefAttr(S.Context, AL));
 }
 
+// Mark alias/ifunc target as used. For C++, we look up the demangled name
+// ignoring parameters. This should handle the majority of use cases while
+// leaveing false positives for namespace scope names and false negatives in
+// the presence of overloads.
+static void markUsedForAliasOrIfunc(Sema &S, Decl *D, const ParsedAttr &AL,
+                                    StringRef Str) {
+  char *Demangled = llvm::itaniumDemangle(Str, /*ParseParams=*/false);
+  if (Demangled)
+    Str = Demangled;
```

Comment:
Doesn't this cause a use-after-free?  This seems wrong to me, with the 'free' below.

Reply:
This test passes asan. 

`itaniumDemangle` allocates memory, which will be freed.

Reply:
Ah, right, my mistake, I thought it was modifying 'Str' as an 'out' parameter.  Disregard.